libraries/chain/controller.cpp

```diff
@@ -1416,12 +1416,14 @@ struct controller_impl {
          // information for those finalizers that don't already have one. This typically should be done when
          // we create the non-legacy fork_db, as from this point we may need to cast votes to participate
          // to the IF consensus. See https://github.com/AntelopeIO/leap/issues/2070#issuecomment-1941901836
-         auto start_block = chain_head; // doesn't matter this is not updated for IRREVERSIBLE, can be in irreversible mode and be a finalizer
-         auto lib_block   = chain_head;
+         block_ref ref = block_handle_accessor::apply<block_ref>(chain_head,
+            overloaded{[&](const block_state_legacy_ptr& head) { return block_ref{head->id(), head->timestamp(), {}}; },
```

Comment:
If we starting before the IF transition, why not use a default safety information that consists of the empty `last_vote` and empty `lock`? That would at least be more consistent then having a malformed block reference with non-empty `block_id` but empty `finality_digest`.

Reply:
done